_posts/2023-08-23-even-friendlier-sql.md

```diff
@@ -0,0 +1,596 @@
+
+If you have not had a chance to read the first installment in this series, please take a quick look [here](https://duckdb.org/2022/05/04/friendlier-sql.html). 
+
+## The future is now
+
+The first few enhancements in this list were included in the “Ideas for the Future” section of the prior post. 
+
+### Reusable column aliases
+
+When working with incremental calculated expressions in a select statement, traditional SQL dialects force you to either write out the full expression for each column or create a common table expression (CTE) around each step of the calculation. Now, any column alias can be reused by subsequent columns within the same select statement. Not only that, but these aliases can be used in the where and order by clauses as well.
+
+#### Old way 1: Repeat yourself
+
+```sql
+select 
+    'These are the voyages of the starship Enterprise...' AS intro,
+    instr('These are the voyages of the starship Enterprise...', 'starship') AS starship_loc
+    substr('These are the voyages of the starship Enterprise...', instr('These are the voyages of the starship Enterprise...', 'starship') + len('starship') + 1) AS trimmed_intro;
+```
+
+#### Old way 2: All the CTEs
+
+```sql
+WITH intro_cte AS (
+    SELECT
+        'These are the voyages of the starship Enterprise...' AS intro
+), starship_loc_cte as (
+    SELECT
+        intro,
+        instr(intro, 'starship') AS starship_loc
+    FROM intro_cte
+)
+SELECT
+    intro,
+    starship_loc,
+    substr(intro, starship_loc + len('starship') + 1) AS trimmed_intro
+FROM starship_loc_cte;
+```
+
+#### New way
+
+```sql
+SELECT 
+     'These are the voyages of the starship Enterprise...' AS intro,
+     instr(intro, 'starship') AS starship_loc,
+     substr(intro, starship_loc + len('starship') + 1) AS trimmed_intro;
+```
+
+|                        intro                        | starship_loc | trimmed_intro |
+|:---|:---|:---|
+| These are the voyages of the starship Enterprise... | 30           | Enterprise... |
+
+
+### Dynamic column selection
+
+Databases typically prefer strictness in column definitions and flexibility in the number of rows. This can help by enforcing data types and recording column level metadata. However, in data science workflows and elsewhere, it is very common to dynamically generate columns (for example during feature engineering).
+
+No longer do you need to know all of your column names up front! DuckDB can select and even modify columns based on regular expression pattern matching, `EXCLUDE` or `REPLACE` modifiers, and even lambda functions (see the [section on lambda functions below](#list-lambda-functions) for details!). 
+
+Let’s take a look at some facts gathered about the first season of Star Trek. Using DuckDB’s [`httpfs` extension](https://duckdb.org/docs/extensions/httpfs), we can query a csv dataset directly from GitHub. It has several columns so let’s `DESCRIBE` it.
+```sql
+install httpfs;
+load httpfs;
```

Comment:
For me, installing the package was enough and loading it happened automatically when running the `CREATE TABLE ... FROM 'https://...'` command. That said, some extensions do not load automatically, so it may be best if we leave the load instruction here.